### PR TITLE
vs/Add test to verify zoom menu buttons functionality and test zoom effect.

### PR DESCRIPTION
**Pull request by vsangereanMOZ**
### Description

Add test to verify zoom menu buttons functionality and test zoom effect.

### Bugzilla bug ID

**Testrail:** https://mozilla.testrail.io/index.php?/cases/view/165064
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1925578

### Type of change

Please delete options that are not relevant.

- [x] New Test
- [ ] New POM
- [ ] Other Changes (Please specify)

### CODE REVIEW

File: modules/page_base.py [hunk 1]
```diff
@@ -711,3 +711,29 @@ def loaded(self):
...
+                return null;
+        """)
+
+        # Parse the transform matrix to extract the scale factor (e.g., matrix(a, b, c, d, e, f))
+        if css_transform_scale:
+            scale_factor = float(css_transform_scale.split('(')[1].split(',')[0])
+            return scale_factor
+
+        # Default return if neither zoom nor transform is set
+        return 1.0
```
Comment (ben-c-at-moz) [inline modules/page_base.py:739]: you could add another method here or in the BOM that does something like (code not guaranteed to work):

```Python
def wait_for_zoom(self, factor: float) -> Page:
    self.expect(lambda _: self.get_css_zoom) == factor)
    return self
```

File: tests/scrolling_panning_zooming/test_zoom_menu_correlation.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    # Open the Firefox Menu panel and prepare to interact with zoom controls
+    panel = PanelUi(driver).open()
+    panel.open_panel_menu()
+
+    # Switch to Firefox's Chrome context for UI interaction
+    with driver.context(driver.CONTEXT_CHROME):
+        # Step 1: Zoom out twice to approximately 90%
+        panel.get_element("zoom-reduce").click()
+        sleep(2)
```
Comment (ben-c-at-moz) [inline tests/scrolling_panning_zooming/test_zoom_menu_correlation.py:30]: You can skip the context switching if you do `panel.click_on("zoom-reduce")`, it's in the BasePage and automatically switches contexts.

File: tests/scrolling_panning_zooming/test_zoom_menu_correlation.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        # Verify that the CSS zoom level reflects the 90% zoom out
+        css_zoom = page.get_css_zoom()
+        assert css_zoom == 0.9, f"Expected CSS zoom == 0.9, but got {css_zoom}"
```
Comment (ben-c-at-moz) [inline tests/scrolling_panning_zooming/test_zoom_menu_correlation.py:34]: You can make this super stable if you implement the `wait_for_zoom` (or whatever you want to name it) mentioned in the page_base commentary

Comment (vsangereanMOZ) [reply tests/scrolling_panning_zooming/test_zoom_menu_correlation.py:34]: Could not get that to work.

**Pull request merged**